DESIGN.md

```diff
@@ -0,0 +1,187 @@
+
+```yaml
+$ kubectl apply -f frontend-component.yaml # create frontend component
+$ kubectl apply -f backend-component.yaml # create backend component
+$ kubectl apply -f application-config.yaml # assign operational traits to components
+```
+
+However, Kubernetes API should not be the end game. Declarative YAML objects are great to build platforms like KubeVela with but when directly exposed to end users, it creates heavy mental burden and high learning curve. For example, even with highest level abstraction, the default [containerized workload schema](https://github.com/oam-dev/spec/blob/master/core/workloads/schema/containerized_workload.md) of OAM still has dozens fields to learn with many rough edges to be aware.
+
+Thus KubeVela intends to introduce a lightweight user facing layer on top of its Kubernetes API objects with following goals in mind:
+
+- Shorten the learning curve of new developers. Most capabilities in KubeVela are developed by big
+companies that run very complex workloads. However, for the bigger developer community, the new user facing layer will provide a much simpler path to on-board KubeVela.
+- Developers can describe their applications and behavior of their components without making assumptions on availability of specific Kubernetes API. For instance, a developer will be able to model auto-scaling needs without referring to the CRD of auto-scaling trait.
+- Provides a single source of truth of the application description. The user facing layer allows developers to work with a single artifact to capture the application definition. This artifact is the definitive truth of how the application is supposed to look like. It simplifies administrative tasks such as change management. It also serves as an anchor for application truth to avoid configuration drifts during operation.
+- Highly extensible. For example, when a new workload type or trait is installed, the end users could access this new capability directly from user interface layer, no re-compile or re-deploy of KubeVela is required.
+
+#### Solution
+
+We concluded above requirements of the user interface layer as introducing a modeling language on top of the existing model objects. With this context, we decide to adopt [CUElang](https://github.com/cuelang/cue) as the modeling language of KubeVela. In detail, every workload or trait definition will inline a cue template as the contract between user and Kubernetes API object.
+
+However, we don't assume users need to learn or write CUElang directly. Instead, the reason we choose CUElang is it's just perfect to help us create developer facing tools based on it. Thus, KubeVela introduced three most common tools based to CUElang for developers to use:
+
+1. A command line tool.
+2. A GUI dashboard.
+3. A Docker Compose style `appfile`.
+
+An example for `vela cli`:
+
+```console
+$ vela comp deploy frontend -t webservice --image oamdev/testapp:v1 --port 80 --app helloworld
```

Comment:
```suggestion
$ vela comp run frontend -t webservice --image oamdev/testapp:v1 --port 80 --app helloworld
```
Currently we use `vela comp run` to create an component. `deploy` also looks nice.

Reply:
I have already refactored to deploy yesteday